internal/pkg/source/sreamer.go

```diff
@@ -0,0 +1,94 @@
+package source
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync"
+	"time"
+)
+
+func (is *Source) StartStreaming(ctx context.Context, send func(msg LazyLogEntries, err error)) {
+	mu := sync.Mutex{}
+	logEntries := make([]LazyLogEntry, 0, 1000)
+	eofEvent := make(chan struct{}, 1)
+	// Load log entries async..
+	go func() {
+		defer func() {
+			eofEvent <- struct{}{}
+		}()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			entry, err := is.ReadLogEntry()
+			if err != nil {
+				if errors.Is(err, io.EOF) {
+					if !is.CanFollow() {
+						return
+					}
+
+					// wait for new log entries to be written to the file,
+					// and try again.
+					ticker := time.NewTicker(200 * time.Millisecond)
```

Comment:
question: why we don't use https://pkg.go.dev/time#NewTimer?
nitpick: let's create a `const` for `200 * time.Millisecond`.